src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import './CreateCourse.css';
+import PropTypes from 'prop-types';
+
+const CreateCourse = ({ authors, setAuthors, onCreateCourse, onCancel }) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [duration, setDuration] = useState('');
+  const [creationDate, setCreationDate] = useState('');
+  const [selectedAuthors, setSelectedAuthors] = useState([]);
+  const [newAuthorName, setNewAuthorName] = useState('');
```

Comment:
Since you have quite a few Inputs in your form, each with its own state, there's 2 ways you could improve this code.
A: Keep the inputs "controlled" via React and create a universal `onChange(key, value)` that would manipulate the state of a certain input based on the key, or event better
B: Make inputs `uncontrolled`, and handle the submission via the form, gathering all the inputs' values at that stage

Please look into these topics if you're interested

Reply:
https://legacy.reactjs.org/docs/uncontrolled-components.html